addons/l10n_in/tests/test_l10n_in_fiscal_position.py

```diff
@@ -105,3 +106,29 @@ def test_l10n_in_fiscal_for_branch(self):
             partner=self.partner_foreign,
             taxes=self.igst_sale_18,
         )
+
+    def test_l10n_in_fiscal_in_bill_to_ship_to(self):
+        self.env.company = self.default_company
+        # Inter State
+        out_invoice = self._assert_invoice_fiscal_position(
+            fiscal_position_ref='fiscal_position_in_inter_state',
+            partner=self.partner_b,
+            taxes=self.igst_sale_18,
+            post=False,
+        )
+        # Intra State
+        out_invoice.write({
+            'l10n_in_state_id': self.env.ref('base.state_in_gj').id,
+        })
+        self.assertEqual(
+            out_invoice.fiscal_position_id,
+            self.env['account.chart.template'].ref('fiscal_position_in_intra_state')
+        )
+        # Outside India (Export/SEZ)
+        out_invoice.write({
+            'l10n_in_state_id': self.env.ref('l10n_in.state_in_oc').id,  # Other Country State
```

Comment:
Shouldn't there be a validation that this "country state" can only be set if `country_id` is not India? And that it always needs to be set to this value in that case?

Reply:
There is a case where OC needs to be set even if the country is India.  And we need this logic for purchases too.

Reply:
![hum](https://media.tenor.com/lD9wxKQYe4YAAAAM/tuesday-morning.gif)

Reply:
And which case is that?
Aren't you getting it confused with `state_in_ot`?

Reply:
![image](https://github.com/user-attachments/assets/e86c093d-1707-4e31-84de-176cbc7bca8d)

Reply:
yes, but Again the answer is in a screenshot :stuck_out_tongue: 
I try to normalize it. where we do events outside India and sell the ticket from India and Attendees is also registered in India but the event is outside India so the place of supply is `Foreign Country`.

If the Odoo India office sale Odoo OXP premium pass to any Indian registered person (Partner) then the Place of supply is `Foreign Country`

![image](https://github.com/user-attachments/assets/9be78222-ac81-4788-9ee9-3b413cd216f7)